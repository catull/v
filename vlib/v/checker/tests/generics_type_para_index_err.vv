fn show_element<T>(arr &T) string {
	return unsafe { '${arr[1]}' }
}

fn main() {
	a := [1, 2, 3]!
	ret := show_element(a)
	println(ret)
}
