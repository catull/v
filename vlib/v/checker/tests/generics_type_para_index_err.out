vlib/v/checker/tests/generics_type_para_index_err.vv:2:24: error: generic type `T` does not support indexing, please use compound type, e.g. `[]T`
    1 | fn show_element<T>(arr &T) string {
    2 |     return unsafe { '${arr[1]}' }
      |                           ~~~
    3 | }
    4 |
